AVsitter2/Plugins/AVcontrol/LockMeister/[AV]LockMeister.lsl

```diff
@@ -0,0 +1,220 @@
+
+            list payload = llParseString2List(msg, ["|"], []);
+
+            if(SITTER != (integer)llList2String(payload, 0)) return; //payload:0 == which seat number this is for.
+
+            if (avatar)
+                ClearChains();
+
+            avatar = id;
+
+            string POSE_NAME = llList2String(payload, 1);
+            integer pose_index = llListFindList(POSES, [POSE_NAME]);
+
+            if(!~pose_index) //If we haven't found a named pose, try to find the wildcard (*) pose.
+            {
+                pose_index = llListFindList(POSES, ["*"]);
+                if(!~pose_index) return;
+            }
+
+            llListenControl(lm_handle,TRUE);
+            llSetTimerEvent(TIMEOUT);
+
+            list data = llCSV2List(llList2String(POSES,pose_index+1));
+            integer i;
+            for(;i<llGetListLength(data);i+=2)
+            {
+                string lm_target = llList2String(data,i);
+                string prim_name = llList2String(data,i+1);
+
+                //lookup the prim linknumber of this anchor point.
+                integer j = llGetNumberOfPrims();
+                while(j--)
+                {
+                    integer link_id = j+1;
+                    if (~llSubStringIndex(llToLower(llGetLinkName(link_id)), prim_name))
+                    {
+                        emitter_links += link_id;
+                        emitter_lmcodes += lm_target;
+                        emitter_states += 0;
+                        llRegionSayTo(avatar,lm_channel,(string)avatar+lm_target);
+                        j=FALSE; //we got what we wanted, bail.
+                    }
+                }
+            }
+        }
+    }
+
+    listen(integer listen_channel, string name, key id, string msg)
+    {
+        if(avatar == NULL_KEY) return;                          //We must have a user.
+        if(llGetOwnerKey(id) != avatar) return;                 //Attachment must belong to user.
+        if((key)llGetSubString(msg,0,35) != avatar) return;     //Target key must be our user.
+
+        if(llGetSubString(msg,-3,-1) == " ok")//lmv1 ' ok' received
+        {
+            //LMV1 Message structure:
+            //Request -> AVATAR_UUIDlmcode
+            //Reply   -> AVATAR_UUIDlmcode ok
+
+            string  lm_target = llGetSubString(msg,36,-4); //strip the 'ok' and the 'target key'.
+            integer emitter_index = llListFindList(emitter_lmcodes,[lm_target]);
+
+            if(!~emitter_index) return;                                 //No emitter configured for this lm code.
+            if(llList2Integer(emitter_states,emitter_index)) return;    //This point already has an lmv1 or lmv2 target.
+
+            SetChain(id,llList2Integer(emitter_links,emitter_index));                           //Draw the chain.
+            emitter_states = llListReplaceList(emitter_states,[1],emitter_index,emitter_index); //Set emitter state to 1 (lmv1 ok)
```

Comment:
Please use `(list)value` instead of `[value]`. It produces shorter code.

Reply:
Done.